### PR TITLE
Implement decorator to handle NaNs in new _helper.py

### PR DESCRIPTION
**Pull request by henrifroese**
- add tests in new test_helpers.py
- decorator is not used anywhere yet, that will be done after this is merged

As discussed last Monday, we will handle the NaNs much simpler than previously suggested by replacing them in the input when the functions don't already handle them on their own (e.g. with `s.str....` we don't need anything else). We believe this decorator is the simplest way to achieve this.

### CODE REVIEW

File: tests/test_helpers.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    tests.addTests(doctest.DocTestSuite(_helper))
+    return tests
+
+
+"""
+Test Decorators.
+"""
+
+
+class TestNLP(PandasTestCase):
```
Comment (jbesomi) [inline tests/test_helpers.py:26]: Rename `TestNLP `  to something more appropriate?

File: tests/test_helpers.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import pandas as pd
```
Comment (jbesomi) [inline tests/test_helpers.py:1]: Initially, I was a bit confused about the goal of this file. Can we add a one-line docstring such as "Unit-test for the `helper` module"

File: texthero/_helper.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
+"""
+Useful helper functions for the texthero library.
+"""
+
+import functools
+import numpy as np
+import warnings
+
+from texthero.preprocessing import fillna
```
[the remark this answers is not in the document]
Comment (henrifroese) [reply texthero/_helper.py:9]: Right, removed.

File: texthero/_helper.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+
+            # Get first input argument (the series) and replace the NaNs.
+            s = args[0]
+            if s.isnull().values.any():
+                warnings.warn(_warning_nans_in_input, Warning)
+                s = s.fillna(value=replace_nans_with)
```
[the remark this answers is not in the document]
Comment (henrifroese) [reply texthero/_helper.py:62]: Agree, done.

File: texthero/_helper.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    """
+
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+
+            # Get first input argument (the series) and replace the NaNs.
+            s = args[0]
+            if s.isnull().values.any():
+                warnings.warn(_warning_nans_in_input, Warning)
```
[the remark this answers is not in the document]
Comment (henrifroese) [reply texthero/_helper.py:61]: Right, `UserWarning` seems most appropriate, changed.

File: tests/test_helpers.py (hunk 1 above)
Comment (henrifroese) [reply tests/test_helpers.py:26]: Wow, not sure how we missed that! :roll_eyes:  It's renamed!

**Comment by henrifroese** [comment]
Implemented the suggested changes.

**Comment by jbesomi** [comment]
👍 👍 👍

**Pull request merged**